openhands/sdk/llm/utils/model_features.py

```diff
@@ -151,3 +160,191 @@ def get_features(model: str) -> ModelFeatures:
+    """
+    separator = "/"
+    split = model.split(separator)
+
+    if len(split) == 1:
+        # no "/" separator found, try with "."
+        separator = "."
+        split = model.split(separator)
+        if split_is_actually_version(split):
+            split = [separator.join(split)]  # undo the split
+
+    if len(split) == 1:
+        matched_provider = ""
+        for provider, models in VERIFIED_MODELS.items():
+            if split[0] in models:
+                matched_provider = provider
+                break
+
+        if matched_provider:
+            return matched_provider, split[0], "/"
+
+        return matched_provider, model, ""
+
+    provider = split[0]
+    model_id = separator.join(split[1:])
+    return provider, model_id, separator
+
+
+def get_unverified_models(
+    aws_region_name: str | None = None,
+    aws_access_key_id: SecretStr | None = None,
+    aws_secret_access_key: SecretStr | None = None,
+) -> dict[str, list[str]]:
+    """
+    Organize a mapping of unverified model identifiers by provider.
+    """
+    result_dict: dict[str, list[str]] = {}
+
+    models = get_supported_llm_models(
+        aws_region_name, aws_access_key_id, aws_secret_access_key
+    )
+    for model in models:
+        provider, model_id, separator = extract_model_and_provider(model)
+        print(provider, model_id, separator)
+
+        # Ignore "anthropic" providers with a separator of "."
+        # These are outdated and incompatible providers.
+        if provider == "anthropic" and separator == ".":
+            continue
+
+        # Dedup verified models
+        if provider in VERIFIED_MODELS and model_id in VERIFIED_MODELS[provider]:
+            continue
+
+        key = provider or "other"
+        if key not in result_dict:
+            result_dict[key] = []
+
+        result_dict[key].append(model_id)
+
+    return result_dict
+
+
+VERIFIED_OPENAI_MODELS = [
+    "gpt-5-2025-08-07",
+    "gpt-5-mini-2025-08-07",
+    "o4-mini",
+    "gpt-4o",
+    "gpt-4o-mini",
+    "gpt-4-32k",
+    "gpt-4.1",
+    "gpt-4.1-2025-04-14",
+    "o1-mini",
+    "o3",
+    "codex-mini-latest",
+]
+
+VERIFIED_ANTHROPIC_MODELS = [
+    "claude-sonnet-4-20250514",
+    "claude-opus-4-20250514",
+    "claude-opus-4-1-20250805",
+    "claude-3-7-sonnet-20250219",
+    "claude-3-sonnet-20240229",
+    "claude-3-opus-20240229",
+    "claude-3-haiku-20240307",
+    "claude-3-5-haiku-20241022",
+    "claude-3-5-sonnet-20241022",
+    "claude-3-5-sonnet-20240620",
+]
+
+VERIFIED_MISTRAL_MODELS = [
+    "devstral-small-2505",
+    "devstral-small-2507",
+    "devstral-medium-2507",
+]
+
+VERIFIED_OPENHANDS_MODELS = [
+    "claude-sonnet-4-20250514",
+    "gpt-5-2025-08-07",
+    "gpt-5-mini-2025-08-07",
+    "claude-opus-4-20250514",
+    "claude-opus-4-1-20250805",
+    "devstral-small-2507",
+    "devstral-medium-2507",
+    "o3",
+    "o4-mini",
+    "gemini-2.5-pro",
+    "kimi-k2-0711-preview",
+    "qwen3-coder-480b",
+]
```

Comment:
I bet we can simplify these into a few regex pattern. Maybe we can move these to `verified_models.py` next to this file though if we are going to a lot here